src/parser.java

```diff
@@ -0,0 +1,179 @@
+package src;
+
+import src.errorHandling.errorPrinter;
+import src.errorHandling.types.Syntax;
+import src.parseTree.nodes.*;
+import src.parseTree.tokens.*;
+
+import java.util.*;
+
+import static src.parseSet.*;
+
+public class parser {
+
+    private static boolean first(Object parent, token token) {
+        boolean result;
+        String parentName = parent.getClass().getSimpleName();
+        if (parent instanceof String) parentName = (String) parent;
+        String tokenName = token.getClass().getSimpleName();
+        if (FIRST.get(parentName) == null || FIRST.get(parentName).get(tokenName) == null) result = false;
+        else result = FIRST.get(parentName).get(tokenName);
+        return result;
```

Comment:
Wall of text formatting for 1, for another, use polymorphism to take either a string or a node along with the token. Then just pass the node through to the string form by using the getClass().getSimpleName()